metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/api/SraXmlRetrieverThroughApi.java

```diff
@@ -18,22 +18,41 @@
 package uk.ac.ebi.ampt2d.metadata.importer.api;
 
 import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
 import org.springframework.retry.annotation.Backoff;
 import org.springframework.retry.annotation.Retryable;
+import org.springframework.web.client.HttpClientErrorException;
 import org.springframework.web.client.RestTemplate;
+import uk.ac.ebi.ampt2d.metadata.importer.ObjectsImporter;
 import uk.ac.ebi.ampt2d.metadata.importer.SraXmlRetrieverByAccession;
 
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
 public class SraXmlRetrieverThroughApi implements SraXmlRetrieverByAccession {
 
-    public static final String ENA_API_URL = "https://www.ebi.ac.uk/ena/data/view/{accessionId}&display=xml";
+    public static final String ENA_API_URL = "https://www.ebi.ac.uk/ena/browser/api/xml/{accessionId}";
+
+    private static final Logger XML_RETRIEVE_LOGGER = Logger.getLogger(ObjectsImporter.class.getName());
 
     private RestTemplate restTemplate = new RestTemplate();
 
-    @Retryable(maxAttemptsExpression="#{${ena.api.attempts}}",
-            backoff=@Backoff(delayExpression="#{${ena.api.delay}}"))
+    @Retryable(maxAttemptsExpression="#{${ena.api.attempts}}", backoff=@Backoff(delayExpression="#{${ena.api.delay}}"))
     @Override
     public String getXml(String accession) {
-        return restTemplate.exchange(ENA_API_URL, HttpMethod.GET, null, String.class, accession).getBody();
+        try {
+            return restTemplate.exchange(ENA_API_URL, HttpMethod.GET, null,
```

Comment:
for simple GET rest calls I think you can just use `restTemplate.get(ENA_API_URL)` which will use sensible default parameters.

Reply:
I don't see a `get()` method in RestTemplate. Did you mean `getForEntity()`?

Otherwise, this makes sense and is indeed simpler. I replaced all `exchange()` calls with `getForEntity()`: https://github.com/EBIvariation/metadata-ws/pull/85/commits/668de0e5d718153f2ab3256b9992775e57861a61

Reply:
yes, I didn't have the doc at hand. There are 2 versions: getForEntity and getForObject. The first returns `ResponseEntity<T>` (`ResponseEntity<String>` in this case) to check http return codes and such. getForObject returns just `T` if you are interested in the content only. https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/client/RestTemplate.html